net.go

```diff
@@ -678,10 +678,15 @@ func (m *Memberlist) rawSendMsgPacket(addr string, node *Node, msg []byte) error
 		}
 		m.nodeLock.RLock()
 		nodeState, ok := m.nodeMap[toAddr]
-		m.nodeLock.RUnlock()
 		if ok {
-			node = &nodeState.Node
+			// if we don't copy this data, we'll be accessing the
+			// pointed-to thing without the mutex held, which means
+			// data races, usually against aliveNode() in state.go.
+			nodeCopy := nodeState.Node
+			node = &nodeCopy
 		}
+		m.nodeLock.RUnlock()
+
```

Comment:
```suggestion
```

Reply:
yeah, probably. the extra blank line wasn't intentional.